pallets/loans/src/lib.rs

```diff
@@ -2026,3 +2030,64 @@ impl<T: Config> LoansRateProvider<AssetIdOf<T>> for Pallet<T> {
         None
     }
 }
+
+#[allow(unused_variables)]
+impl<T: Config> LoansOperator<AssetIdOf<T>, AccountIdOf<T>, BalanceOf<T>> for Pallet<T> {
+    fn do_mint(
+        supplier: &AccountIdOf<T>,
+        asset_id: &AssetIdOf<T>,
+        amount: BalanceOf<T>,
+    ) -> Result<(), DispatchError> {
+        Ok(())
+    }
+    fn do_borrow(
+        borrower: &AccountIdOf<T>,
+        asset_id: &AssetIdOf<T>,
+        amount: BalanceOf<T>,
+    ) -> Result<(), DispatchError> {
+        Ok(())
+    }
+    fn do_collateral_asset(
+        supplier: &AccountIdOf<T>,
+        asset_id: &AssetIdOf<T>,
+        enable: bool,
+    ) -> Result<(), DispatchError> {
+        Ok(())
+    }
+    fn do_repay_borrow(
+        borrower: &AccountIdOf<T>,
+        asset_id: &AssetIdOf<T>,
+        amount: BalanceOf<T>,
+    ) -> Result<(), DispatchError> {
+        Ok(())
+    }
+    fn do_redeem(
+        supplier: &AccountIdOf<T>,
+        asset_id: &AssetIdOf<T>,
+        amount: BalanceOf<T>,
+    ) -> Result<(), DispatchError> {
+        Ok(())
+    }
+}
+
+#[allow(unused_variables)]
+impl<T: Config> LoansMarketDataProvider<AssetIdOf<T>, BalanceOf<T>> for Pallet<T> {
+    fn get_market_info(asset_id: &AssetIdOf<T>) -> Result<MarketInfo, DispatchError> {
```

Comment:
MarketInfo  def can be removed, we can return the whole Market

Reply:
first try this way but the market structure is huge and including some domain specific impl including `JumpModel` ,not sure if it is a better way